### DISCUSSION TITLE
Use different output.libraryTarget for a Web Worker

### DISCUSSION THREAD

**Discussion by michal-perlakowski** [webpack/webpack#14853] [Q&A]
I have this Webpack configuration:

```
{
  output: {
    libraryTarget: "system",
    ...
  },
  ...
}
```

I'm trying to use a Web Worker. I'm using the [standard Webpack 5 syntax](https://webpack.js.org/guides/web-workers/):

```
new Worker(new URL('./MyWorker', import.meta.url));
```

Now Webpack outputs the Web Worker as a System.js module. This obviously doesn't work, because `System` isn't defined in the Worker scope. How can I change it to something different, like ES module, without affecting the main bundle?

The only workaround I've found is returning an array with two separate configurations in `webpack.config.js`, which isn't very convenient.

**Comment by alexander-akait** [comment]
Sorry, only multi compiler mode right now
